libraries/AP_HAL_ChibiOS/RCOutput_bdshot.cpp

```diff
@@ -496,6 +496,11 @@ __RAMFUNC__ void RCOutput::bdshot_finish_dshot_gcr_transaction(virtual_timer_t*
 #ifdef HAL_GPIO_LINE_GPIO56
     TOGGLE_PIN_DEBUG(56);
 #endif
+    if (group->dshot_waiter == nullptr) {   // transaction was cancelled, leave everything alone
```

Comment:
i think this change won't work, when the bug happens 'this' is nullptr, so we are checking an incorrect memory location

Reply:
No, I don't think so. The "this" is a pointer set in the timer, it cannot be null unless the timer is reset in which case the callback is not called. The nullptr is not "this" but this->dshot_waiter which *was* so to nullptr outside of a lock hence the bug.

Reply:
Incidentally I don't believe this can ever be triggered because we now cancel the timer inside a lock together with setting the value to nullptr

Reply:
you're right, sorry!